packages/ui/src/components/Card.tsx

```diff
@@ -41,9 +43,14 @@ function getQAPromptContents<PT extends QAPromptTask | ApplicationPromptTask>(
       const applicationReviewItem = reviewItem as ReviewAreaItem<
         ApplicationPromptTask
       >;
-      return applicationReviewItem.prompt.variants[
-        applicationReviewItem.taskParameters.variantIndex
-      ];
+      const taskParameters = applicationReviewItem.taskParameters as {
+        variantIndex: number;
+      };
+      const applicationPrompt = applicationReviewItem.prompt as ApplicationPrompt;
+
+      return applicationPrompt.variants[taskParameters.variantIndex];
+    case clozePromptType:
```

Comment:
Odd—you shouldn't have to do this, since the generic constraint should exclude clozes! I'm guessing that the compiler couldn't figure that out?

Reply:
Yep this unfortunate. It looks like through bumping the package the code below is giving typescript trouble across the board.
 
```typescript
export type PromptTaskParametersOf<
  PT extends PromptTask
> = PT extends AbstractPromptTask<any, any, infer PTP> ? PTP : never;
```

Typescript seems unable to coerce the type any longer (see screenshot) which prevents it from detecting that this case is unreachable.

<img width="598" alt="Screen Shot 2021-05-18 at 2 09 31 PM" src="https://user-images.githubusercontent.com/8084674/118724140-1902b880-b7e3-11eb-9a74-e4a7a105a579.png">

Without including that switch case the following error occurs:

```bash
$ tsc -b packages/tsconfig.json
packages/ui/src/components/Card.tsx:38:4 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.

38 ): QAPromptContents {
      ~~~~~~~~~~~~~~~~
```

Adding the cloze case resolves the issue.

I am worried this lack of type information will deteriorate type safety across the board. Any thoughts?